backend/lib/services/infrastructureSecrets.js

```diff
@@ -248,27 +250,33 @@ exports.patch = async function ({ user, namespace, bindingName, body }) {
   })
 }
 
-exports.remove = async function ({ user, namespace, bindingName }) {
+exports.remove = async function ({ user, namespace, name }) {
   const client = user.client
 
-  const secretBinding = await client['core.gardener.cloud'].secretbindings.get(namespace, bindingName)
+  const secretBinding = await client['core.gardener.cloud'].secretbindings.get(namespace, name)
   const secretName = _.get(secretBinding, 'secretRef.name')
+  const secretNamespace = _.get(secretBinding, 'secretRef.namespace')
 
-  checkIfOwnSecret(secretBinding)
+  if (!isOwnSecret(secretBinding)) {
+    throw new MethodNotAllowed('Remove allowed only for secrets in own namespace')
+  }
 
   const { items: shootList } = await shoots.list({ user, namespace })
-  const predicate = (item) => {
-    const itemSecretBindingName = _.get(item, 'spec.secretBindingName')
-    return itemSecretBindingName === bindingName
-  }
-  const secretReferencedByShoot = _.find(shootList, predicate)
+  const secretReferencedByShoot = _.find(shootList, ['spec.secretBindingName', name])
```

Comment:
Not part of this PR but also PreconditionFailed is not used correctly here. All of these erros should be 422.

Reply:
✅